### ISSUE TITLE
Code doesn't attach to node process in Javascript Debug Terminal

### ISSUE THREAD

**Issue by jbree** [microsoft/vscode#102404]
- VSCode Version: 1.47
- OS Version: Windows 10 v1803 (17134.1550)

Steps to Reproduce:

1. Create a new Javascript Debug Terminal. Using git-bash/mingw as my shell.
2. Set a breakpoint in code
3. Run node process, e.g. `npm t` in debug terminal
4. Process launches, but sits idle after output similar to the following:
> Debugger listening on ws://127.0.0.1:54596/2827d3c1-baf0-499c-9e72-f964b7f76038
> For help, see: https://nodejs.org/en/docs/inspector

Does this issue occur when all extensions are disabled?: Yes

**Additional Info:**

- Node version v12.14.1
- Also tried this on Insiders, using PowerShell instead of git-bash/mingw, saw the same result.
- Tried downgrading to 1.46.1, but after the upgrade to 1.47 it was broken on the old version too.
- Tried setting `"debug.javascript.usePreview"` setting to both `true` and `false` (relaunch vscode after change). Fails on both.

**Comment by connor4312** [comment]
What's your current setting for debug.node.autoAttach?

**Comment by jbree** [comment]
I haven't modified that setting directly in the settings JSON, but using the toggle in the bottom status bar I've tried both enabled and disabled with the same results.

**Comment by jbree** [comment]
One additional observation: When running 1.47, I see the initial

>Debugger listening on ws://127.0.0.1:55174/21ff3d25-4787-443c-8421-2229880e0536
>For help, see: https://nodejs.org/en/docs/inspector
Debugger listening on ws://127.0.0.1:55175/dc791721-e25a-4fc6-b630-98e113fb9847
For help, see: https://nodejs.org/en/docs/inspector

**Comment by jbree** [comment]
Sorry, closed by mistake.

**Comment by connor4312** [comment]
Okay, try going in your user settings and set it to "disabled", then reopening your terminal. There's an issue there that will be fixed in the next vscode release (est. Thursday).

**Comment by jbree** [comment]
Setting `"debug.node.autoAttach": "disabled"` had no effect--in fact, even after reload it didn't cause the status bar indicator to go away as the documentation suggests it should.

I tried disabling the builtin vscode.debug-auto-launch extension, and that also didn't cause any change in the issue.

**Comment by connor4312** [comment]
Okay, I know what's happening. I believe this should fix it for you:

1. Set `"debug.node.autoAttach": off`
1. Reload the window
1. Set `"debug.node.disabled"`. The badge should then disappear.
1. Wait a few seconds. Then opening the terminal should work.

**Comment by jbree** [comment]
Followed instructions; "debug.node.disabled" shows up as an unknown configuration setting, but I tried anyways. Auto Attach badge did not go away.

Also tried "debug.node.autoAttach": "disabled" in case that's what you meant, and that also had no effect.

**Comment by connor4312** [comment]
Huh, okay. Do any errors appear in the console if you run the `Open Developer Tools` command?

**Comment by jbree** [comment]
These errors come up before I open the javascript debug terminal or try to execute a command therein, but I'll paste anyways:

```
console.ts:137 [Extension Host] (node:2140) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
_doInvokeHandler @ rpcProtocol.ts:402
_invokeHandler @ rpcProtocol.ts:387
_receiveRequest @ rpcProtocol.ts:303
_receiveOneMessage @ rpcProtocol.ts:230
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:453
_receiveMessage @ ipc.net.ts:733
(anonymous) @ ipc.net.ts:592
fire @ event.ts:587
acceptChunk @ ipc.net.ts:239
(anonymous) @ ipc.net.ts:200
t @ ipc.net.ts:28
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166

console.ts:137 [Extension Host] (node:2140) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
t.log @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
_doInvokeHandler @ rpcProtocol.ts:402
_invokeHandler @ rpcProtocol.ts:387
_receiveRequest @ rpcProtocol.ts:303
_receiveOneMessage @ rpcProtocol.ts:230
(anonymous) @ rpcProtocol.ts:105
fire @ event.ts:587
fire @ ipc.net.ts:453
_receiveMessage @ ipc.net.ts:733
(anonymous) @ ipc.net.ts:592
fire @ event.ts:587
acceptChunk @ ipc.net.ts:239
(anonymous) @ ipc.net.ts:200
t @ ipc.net.ts:28
emit @ events.js:203
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:276
Readable.push @ _stream_readable.js:210
onStreamRead @ internal/stream_base_commons.js:166
```

**Comment by connor4312** [comment]
Okay, thanks. This should be fixed in the upcoming recovery release (est. Thursday) and the next Insiders version. You'll have to wait a couple days to use the debug terminal, sorry!

**Comment by jbree** [comment]
No worries, thanks for the support!

**Comment by connor4312** [comment]
I found some better mitigation steps here if you'd like to try them: https://github.com/microsoft/vscode/issues/102057#issuecomment-657728467

**Comment by jbree** [comment]
Tried v1.47.2 and did not have success. I got curious and wondered if it was a problem with my workspace. I recreated my workspace and the debugger worked.

To try to isolate the issue, I went as far as to delete all of the settings in the old `*.code-workspace` to make it match the newly created one, and even that didn't make debugging work in the old one. So, maybe something cached behind the scenes that I don't know about

**TL;DR: Created a new workspace, debug terminal works there**

**Comment by connor4312** [comment]
Thanks for following up! Yes auto attach does cache some stuff per-workspace. New 1.47.2 workspaces should be totally fine. Workspaces opened in 1.47.0-1 _may_ need to you toggle auto attach to disabled (and then back on if you want) to get cleared out.

**Comment by jbree** [comment]
Ah yep, that did the trick to fix my old workspace. Thanks for your work on this!

**Comment by SinGlEBW** [comment]
For some reason, the restart in console stopped working. I had to abandon debug.node.autoAttach: "off". I use the following

   {  
         "name": "nodemon",
         "type": "node",
         "request": "launch",
         "runtimeExecutable": "nodemon",
         "program": "${workspaceFolder}/server/express.js",
         "restart": true,
         "console": "integratedTerminal",
         "autoAttachChildProcesses": true,
         "internalConsoleOptions": "neverOpen",
         "skipFiles": [
            "<node_internals>/**"
         ],
      }

+ open new Terminal and run React   -  npm start

**Comment by israelpgjm** [comment]
### y que no le da el gusto que le gusta el que

**Comment by connor4312** [comment]
> the restart in console stopped working

I'm not sure what you mean by this. 

>  I had to abandon debug.node.autoAttach: "off"

If you need to clear variables, make set `debug.node.autoAttach` to `disabled`, and then you should be able to turn it back on (may need to restart terminals or reload the window after doing so)
